react/components/LoginLogout.tsx

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable no-console */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import { LoginAccountDialog } from 'extensions.account-dialog'
+import React, { Fragment, useEffect, useState } from 'react'
+import { useCssHandles } from 'vtex.css-handles'
+
+import { deleteCookieByName, getCookieByName } from '../utils'
+
+const COOKIE_LOGIN_EXP = 'VtexAppsLoginExpInMs'
+
+interface Props {
+  loginText: string
+  logoutText: string
+}
+
+declare global {
+  interface Window {
+    __RUNTIME__: {
+      workspace?: string
+    }
+  }
+}
+
+const isUserLoggedIn = () => {
+  const exp = getCookieByName(COOKIE_LOGIN_EXP)
+  return !!exp && parseInt(exp, 10) > Date.now()
+}
+
+const CSS_HANDLES = ['styledLoginLogout'] as const
+function LoginLogout({ loginText, logoutText }: Props) {
+  const [loggedIn, setLoggedIn] = useState<boolean | null>(null)
+  const [startedLogin, setStartedLogin] = useState<boolean>(false)
```

Comment:
Can we merge these statuses into one? It could be easier to scale
```suggestion
  const [loginStatus, setLoginStatus] = useState<'unknown' | 'authenticating' | 'authenticated' | 'unauthenticated'>('unknown')
```

Reply:
Sounds better, thanks!